Projectoid.js

```diff
@@ -0,0 +1,292 @@
+      throw "Error: Access Token Not Found";
+      return;
+    }
+  }
+
+  let data = {
+    method: "post",
+    path: "telegram/botpanel/adduser.php",
+    body: {
+      bot_id: bot.token.split(":")[0],
+      user_id: userid,
+      access_token: access_token,
+    },
+    onSuccess: !options.command ? null : options.command,
+  };
+  apiCall(data);
+}
+
+function broadcast(options) {
+  if (!options) {
+    throw "Error: broadcast: options not found";
+    return;
+  }
+  if (!options.method) {
+    throw "Error: broadcast: options not found";
+    return;
+  }
+  if (options.method == "forwardBroadcast") {
+    forwardBroadcast(options);
+    return;
+  }
+  if (options.method == "copyBroadcast") {
+    copyBroadcast(options);
+    return;
+  }
+
+  var text = !options.text ? null : options.text;
+  var type = !options.type ? null : options.type;
+  var file_id = !options.file_id ? null : options.file_id;
+  var caption = !options.caption ? null : options.caption;
+  var parseMode = !options.parseMode ? null : options.parseMode;
+  var disableWebPreview = !options.disableWebPreview
+    ? null
+    : options.disableWebPreview;
+  var protectContent = !options.protectContent ? false : options.protectContent;
+  var webhookUrl = !options.webhookUrl ? null : options.webhookUrl;
+  var cmd = !options.command ? null : options.command;
+
+  if (!options.access_token) {
+    var access_token = Bot.getProperty("Projectoid_AccessToken");
+    if (!access_token) {
+      throw "Error: broadcast: Bot Access Token Not Found";
+      return;
+    }
+  }
+
+  let data = {
+    path: "telegram/botpanel/broadcast.php",
+    method: "post",
+    body: {
+      bot: bot.name,
+      bot_token: bot.token,
+      access_token: access_token,
+      admin: user.telegramid,
+      method: options.method,
+      text: text,
+      type: type,
+      file_id: file_id,
+      caption: caption,
+      parseMode: parseMode,
+      disableWebPreview: disableWebPreview,
+      protectContent: protectContent,
+      webhookUrl: webhookUrl,
+    },
+    onSuccess: cmd,
+  };
+  apiCall(data);
+}
+
+function forwardBroadcast(options) {
+  if (!options) {
+    throw "Error: forwardBroadcast: options not found";
+    return;
+  }
+  var from_chat_id = options.from_chat_id;
+  var message_id = options.message_id;
+
+  if (!from_chat_id || !message_id) {    
+    throw "Error: forwardBroadcast: chat id or message id was not found";
+    return;
+  }
+  var protectContent = !options.protectContent ? false : options.protectContent;
+  var webhookUrl = !options.webhookUrl ? null : options.webhookUrl;
+  var cmd = !options.command ? null : options.command;
+
+  if (!options.access_token) {
+    var access_token = Bot.getProperty("Projectoid_AccessToken");
+    if (!access_token) {
+      throw "Error: forwardBroadcast: Bot Access Token Not Found";
+      return;
+    }
+  }
+
+  let data = {
+    path: "telegram/botpanel/broadcast.php",
+    method: "post",
+    body: {
```

Comment:
body: {
      access_token: ,
      bot_token: bot.token,


it is enough

Reply:
you mean we don't need path and methods?

Reply:
try to ask ChatGPT here